theano/tensor/nnet/sigm.py

```diff
@@ -602,6 +602,7 @@ def local_exp_over_1_plus_exp(node):
             else:
                 # case: 1/(1+exp(x))
                 sigmoids.append(sigmoid(-t))
+            copy_stack_trace(node.outputs[0], sigmoids[-1])
```

Comment:
So, that is the `Canonizer` class in `tensor/opt.py` that is responsible for that.
It is a big piece, so for now you can just disable this test and open a ticket, or mention it in an existing ticket.